zone_cache_reserve.go

```diff
@@ -0,0 +1,79 @@
+package cloudflare
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"time"
+)
+
+var validOnOffSettingValues = []string{"on", "off"}
+
+// ZoneCacheReserve is the structure of the API object for the
+// cache reserve setting.
+type ZoneCacheReserve struct {
+	ID         string    `json:"id,omitempty"`
+	ModifiedOn time.Time `json:"modified_on,omitempty"`
+	Value      string    `json:"value"`
+}
+
+// CacheReserveDetailsResponse is the API response for the
+// cache reserve setting.
+type CacheReserveDetailsResponse struct {
+	Result ZoneCacheReserve `json:"result"`
+	Response
+}
+
+type updateZoneCacheReserveRequest struct {
+	Value string `json:"value"`
+}
+
+type zoneCacheReserveSingleResponse struct {
+	Response
+	Result ZoneCacheReserve `json:"result"`
+}
+
+// GetZoneCacheReserve returns information about the current cache reserve settings
+//
+// API reference: https://developers.cloudflare.com/api/operations/zone-cache-settings-get-cache-reserve-setting
+func (api *API) GetZoneCacheReserve(ctx context.Context, zoneID string) (ZoneCacheReserve, error) {
```

Comment:
have a look at https://github.com/cloudflare/cloudflare-go/blob/master/docs/experimental.md for how we are managing method signatures going forward for a consistent experience. #1265 has some prior art for these method signatures if you'd like something to lean on.

Reply:
I fail to see how I can map this to the service pattern described in experimental as cache reserve is not exposed as a service/entity on its own but more like a setting tied to a zone

Would it make sense to implement it (mostly move the code from the PR) as methods in the zone.go file? Similarly to the Zone SSL Settings methods here https://github.com/cloudflare/cloudflare-go/blob/10cc0fdc82ef5308226a1273f1f769f0de92c7eb/zone.go#L826-L858

Reply:
the entity/service/relationships portion of this doesn't really matter as every endpoint is essentially it's own entity and should be treated as such. the intent is to get the definition and accessing the entity the same across the board (which is what the experimental client is about).

i've gone ahead and updated this PR with the experimental client signatures. in future, if we decide to also expose this as a zone level setting, we can explore that interface at another time.